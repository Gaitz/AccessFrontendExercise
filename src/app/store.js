import { configureStore } from "@reduxjs/toolkit";
import userListSlice from "../features/list/userListSlice";
import userDetailSlice from "../features/detail/userDetailSlice";

const store = configureStore({
  reducer: { list: userListSlice, detail: userDetailSlice },
});

export default store;
