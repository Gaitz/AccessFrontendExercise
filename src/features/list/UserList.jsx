import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { getGitHubUsers, appendPageIndex, PER_PAGE } from "./userListSlice";
import styles from "./UserList.module.css";
import { Link } from "react-router-dom";
import { PageTitle, Pending } from "../../shared/component";

const UserList = () => {
  const dispatch = useDispatch();

  const users = useSelector((state) => state.list.users);
  const pageIndex = useSelector((state) => state.list.pageIndex);
  const isPending = useSelector((state) => state.list.isPending);

  useEffect(() => {
    dispatch(getGitHubUsers(pageIndex));
  }, [pageIndex, dispatch]);

  const anUserInfo = (user, order) => {
    return (
      <li className={styles.anUserInfo} key={user.id}>
        <Link to={`/${user.login}`}>
          <p>{order}</p>
          <div className={styles.avatar}>
            <img src={user.avatarUrl} alt="avatar" />
          </div>
          <p className={styles.userLogin}>{user.login}</p>
          <p className={styles.siteAdmin}>
            Site Admin: {user.siteAdmin ? "true" : "false"}
          </p>
        </Link>
      </li>
    );
  };

  return (
    <article className={styles.listComponent}>
      <PageTitle>GitHub User List</PageTitle>
      {isPending === true ? (
        <Pending />
      ) : (
        <>
          <div className={styles.paginatedBlock}>
            <p>
              This page contains {users.length} items, Current page index:
              {pageIndex + 1}
            </p>
            <button
              onClick={(event) => {
                event.preventDefault();
                dispatch(appendPageIndex(1));
              }}
            >
              Get Next Page
            </button>
          </div>
          <ul className={styles.userList}>
            {users.map((user, order) =>
              anUserInfo(user, pageIndex * PER_PAGE + order + 1)
            )}
          </ul>
        </>
      )}
    </article>
  );
};

export default UserList;
