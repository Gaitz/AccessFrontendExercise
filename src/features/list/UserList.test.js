import React from "react";
import { Provider } from "react-redux";
import store from "../../app/store";
import { render } from "@testing-library/react";
import UserList from "./UserList";

test("snapshot UserList", () => {
  const { container } = render(
    <Provider store={store}>
      <UserList></UserList>
    </Provider>
  );
  expect(container).toMatchSnapshot();
});
