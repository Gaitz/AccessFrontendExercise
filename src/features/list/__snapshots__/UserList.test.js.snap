// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot UserList 1`] = `
<div>
  <article
    class="listComponent"
  >
    <h2
      class="pageTitle"
    >
      GitHub User List
    </h2>
    <p
      class="pendingBlock"
    >
      Waiting for request 
    </p>
  </article>
</div>
`;
